src/SixLabors.Fonts/EncodingExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+#if !SUPPORTS_ENCODING_STRING
+using System;
+using System.Text;
+
+namespace SixLabors.Fonts
+{
+    /// <summary>
+    /// Extension methods for the <see cref="Encoder"/> type.
+    /// </summary>
+    internal static unsafe class EncodingExtensions
+    {
+        /// <summary>
+        /// Gets a string from the provided buffer data.
+        /// </summary>
+        /// <param name="encoding">The encoding.</param>
+        /// <param name="buffer">The buffer.</param>
+        /// <returns>The string.</returns>
+        public static string GetString(this Encoding encoding, ReadOnlySpan<byte> buffer)
```

Comment:
The `ReadString` and `ReadTag` methods in `BigEndianBinaryReader` could maybe make use of this extension method